### ISSUE TITLE
modules don't directly depend on core

### ISSUE THREAD

**Issue by yschimke** [rsocket/rsocket-java#498]
http://central.maven.org/maven2/io/rsocket/rsocket-transport-netty/0.11.0/rsocket-transport-netty-0.11.0.pom

Clients need to bring in both dependencies.  Is this deliberate?

**Comment by mostroverkhov** [comment]
@yschimke module in question has implementation dependency on `core`. More interesting thing is `reactor-netty` is api dependency, so likely it's typo - has to be vice versa, as  transports themselves can be used without `reactor-netty` on compile classpath

**Comment by mostroverkhov** [comment]
Apparently parent module has to be published aswell

**Comment by nebhale** [comment]
I’ll chase this down today.

**Comment by nebhale** [comment]
Agreed on the latter.  Artisinally handcrafting some free range POMs now.

**Comment by nebhale** [comment]
@mostroverkhov I think the transport backing libraries are going to need to be `compile` dependencies as well.  Both [Netty](https://github.com/rsocket/rsocket-java/blob/1.0.x/rsocket-transport-netty/src/main/java/io/rsocket/transport/netty/server/TcpServerTransport.java#L48) and [Aeron](https://github.com/rsocket/rsocket-java/blob/1.0.x/rsocket-transport-aeron/src/main/java/io/rsocket/aeron/server/AeronServerTransport.java#L38) expose their implementing types in their APIs.

**Comment by mostroverkhov** [comment]
But also can create using [these](https://github.com/rsocket/rsocket-java/blob/5fedb1d83d3ab1dfafa9e7a1b29768088a0e026a/rsocket-transport-netty/src/main/java/io/rsocket/transport/netty/server/TcpServerTransport.java#L38) ? But given that `rsocket-core` is built on `reactor` and uses some bits of `netty`, there is indeed no big reason to have `reactor-netty` as runtime dependency.

**Comment by yschimke** [comment]
We should probably get the gradle ones right first. You can’t use a transport with core, it is part of their public API so gradle api instead of implementation in probably right.

**Comment by nebhale** [comment]
I'll attach this to the PR as well, but here's a [sample of the POMs](https://gist.github.com/nebhale/668720e6fff89df77424f03d731a79bc) as they are created now.

**Comment by nebhale** [comment]
@yschimke pending your review the last couple of releases that have gone out, should solve this issue.
